### PR TITLE
Self-copy fixes

### PR DESCRIPTION
**Pull request by tbennun**
In transformations and memlet creation

### CODE REVIEW

File: dace/memlet.py [hunk 1]
```diff
@@ -362,11 +362,21 @@ def try_initialize(self, sdfg: 'dace.sdfg.SDFG',
...
+        is_data_dst = False
+        if isinstance(path[0].src, AccessNode):
+            if path[0].src.data == self._data:
+                is_data_src = True
         if isinstance(path[-1].dst, AccessNode):
             if path[-1].dst.data == self._data:
-                is_data_src = False
-        self._is_data_src = is_data_src
+                is_data_dst = True
+        if is_data_src and is_data_dst:
```
Comment (alexnick83) [inline dace/memlet.py:373]: Is this potentially also causing us issues, i.e., do we need to know with great confidence the data source?

File: dace/memlet.py [hunk 2]
```diff
@@ -547,7 +557,10 @@ def _label(self, shape):
             result += ' (CR: %s)' % wcrstr
 
         if self.other_subset is not None:
-            result += ' -> [%s]' % str(self.other_subset)
+            if self._is_data_src is False:
+                result += ' <- [%s]' % str(self.other_subset)
```
Comment (alexnick83) [inline dace/memlet.py:561]: Quite smart.

File: dace/transformation/dataflow/redundant_array.py [hunk 3]
```diff
@@ -1326,21 +1326,22 @@ def gnode(nname):
                 # A -- (0, a:b)/(c:c+b) --> V -- (c+d)/None --> others
                 # A -- (0, a+d)/None --> others
                 e3.data.data = in_array.data
+                e3.data._is_data_src = True
```
Comment (alexnick83) [inline dace/transformation/dataflow/redundant_array.py:1329]: I see now; so it is indeed an issue with whether we can really tell which is the data source.

Comment (alexnick83) [reply dace/transformation/dataflow/redundant_array.py:1329]: Btw, this formula might be also needed in the other transformations too. I would put a note for now.

**Review by alexnick83 — APPROVED** [review]
LGTM. However, this PR does not address the general issue where self-copies (in frontend code) should produce actual data copies with a transient (and not views) in the SDFG. Right?

File: dace/transformation/dataflow/redundant_array.py (hunk 3 above)
Comment (tbennun) [reply dace/transformation/dataflow/redundant_array.py:1329]: when we add the edge it should be set automatically. in self-copies it was ambiguous though and it reset it. This line may not be necessary

**Pull request merged**